components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java

```diff
@@ -48,6 +49,9 @@ public URI getUri() throws URISyntaxException {
         } else {
             Objects.requireNonNull(urlTemplate);
             Objects.requireNonNull(queryParameters);
+            if(urlTemplate.toLowerCase().contains("{+baseurl}") && !pathParameters.containsKey("baseurl"))
```

Comment:
```suggestion
            if(!pathParameters.containsKey("baseurl") && urlTemplate.toLowerCase().contains("{+baseurl}"))
```
to avoid checking the string when we already have an entry